crates/epaint/src/tessellator.rs

```diff
@@ -1067,19 +1123,22 @@ fn stroke_path(
         if thin_line {
             // Fade out thin lines rather than making them thinner
             let radius = feathering / 2.0;
-            let color = mul_color(stroke.color, stroke.width / feathering);
-            if color == Color32::TRANSPARENT {
-                return;
+            if let ColorMode::Solid(color) = stroke.color {
+                let color = mul_color(color, stroke.width / feathering);
+                if color == Color32::TRANSPARENT {
+                    return;
+                }
             }
             for p in path {
+                let color = mul_color(get_color(&stroke.color, p.pos), stroke.width / feathering);
                 out.colored_vertex(p.pos + radius * p.normal, color);
                 out.colored_vertex(p.pos - radius * p.normal, color);
             }
         } else {
             let radius = stroke.width / 2.0;
             for p in path {
-                out.colored_vertex(p.pos + radius * p.normal, stroke.color);
-                out.colored_vertex(p.pos - radius * p.normal, stroke.color);
+                out.colored_vertex(p.pos + radius * p.normal, get_color(&stroke.color, p.pos));
+                out.colored_vertex(p.pos - radius * p.normal, get_color(&stroke.color, p.pos));
```

Comment:
The new color code has the potential of slowing down this function.

I'd like you to add a benchmark to `crates/epaint/benches/benchmark.rs` to test this for two cases:
* thin solid line (stroke width < feathering)
* thick solid line (stroke width > feathering)

Reply:
resolved with c9efc240d03842e16716a92d1077d84ee06e7543